preset/kaleidoscope/kaleidoscope.cc

```diff
@@ -145,7 +145,10 @@ void Kaleidoscope::OnDrawFrame(absl::Span<const float> samples,
   LOG(DEBUG) << "Got texture size location: " << texture_size_location;
   glUniform2i(texture_size_location, width(), height());
 
-  glViewport(0, 0, width(), height());
+  auto longer_dim = std::max(width(), height());
```

Comment:
s/longer_dim/longer_dimension/

Could you also plumb a boolean flag through for selecting whether or not to perform aspect ratio correction? If not in this PR, please add a `TODO(<issue #>): <small description of what needs to be done>`.

Also, because `glViewport` is only adjusted before the final draw call that puts the pixels into the window buffer, we will be supersampling when the buffer is not square. This results in loss of detail. Perhaps a better strategy is to render the preset on a square raster, and then crop for display?